sql-indent-left.el

```diff
@@ -0,0 +1,189 @@
+;;
+;; This program is free software; you can redistribute it and/or
+;; modify it under the terms of the GNU General Public License as
+;; published by the Free Software Foundation; either version 3, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+;; General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program; see the file COPYING.  If not, write to
+;; the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+;; Floor, Boston, MA 02110-1301, USA.
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;;; Code:
+
+(require 'sql-indent)
+
+(defun sqlind-adjust-operator-left (_syntax base-indentation)
+  "Adjust the indentation for operators in select clauses.
+
+Select columns are lined up to the operands, not the operators.
+For example.
+
+    select col1, col2
+              || col3 as composed_column,
+           col4
+        || col5 as composed_column2
+    from   my_table
+    where  cond1 = 1
+    and    cond2 = 2;
+
+This is an indentation adjuster and needs to be added to the
+`sqlind-indentation-offsets-alist`"
+  (save-excursion
+    (back-to-indentation)
+    ;; If there are non-word constituents at the beginning if the line,
+    ;; consider them an operator and line up the line to the first word of the
+    ;; line, not the operator
+    (cond ((looking-at "\\W+")
+	   (let ((ofs (length (match-string 0))))
+	     (forward-line -1)
+	     (end-of-line)
+	     (sqlind-backward-syntactic-ws)
+	     (forward-sexp -1)
+	     (- (current-column) ofs)))
+	  ('t base-indentation))))
+
+(defun sqlind-adjust-operator-right (_syntax base-indentation)
+  "Adjust the indentation for operators in select clauses.
+
+Select columns are lined up to the operands, not the operators.
+For example.
+
+  select col1, col2
+            || col3 as composed_column,
+         col4
+      || col5 as composed_column2
+    from my_table
+   where cond1 = 1
+     and cond2 = 2;
+
+This is an indentation adjuster and needs to be added to the
+`sqlind-indentation-offsets-alist`"
+  (save-excursion
+    (back-to-indentation)
+    ;; If there are non-word constituents at the beginning if the line,
+    ;; consider them an operator and line up the line to the first word of the
+    ;; line, not the operator
+    (cond ((looking-at "\\W+")
+	   (let ((ofs (length (match-string 0))))
+	     (forward-line -1)
+	     (end-of-line)
+	     (sqlind-backward-syntactic-ws)
+	     (forward-sexp -1)
+	     (- (current-column) ofs)))
+	  ((looking-at "\\(?:and\\|or\\|not\\)")
+	   (let ((ofs (length (match-string 0))))
+	     (+ base-indentation (- 5 ofs))))
+	  ('t base-indentation))))
+
+(defun sqlind-lone-semicolon (syntax base-indentation)
```

Comment:
I think this function could be added to `sql-indent.el`, as it seems generic enough, and I already added some other indentation functions which are not directly used, but had to add them to implement indentation rules for the Emacs SQL mode maintainer.

Can you just add an entry to customize-indentation.md. I plan to re-design that documentation, so just add a simple bullet point to remind me to document it. Thanks.